File: app/src/main/java/com/example/android/kannada/ColorsActivity.java
package com.example.android.kannada;

import android.media.MediaPlayer;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ListView;

import java.util.ArrayList;

/**
 * Created by nsethi on 03-Jul-16.
 */
public class ColorsActivity extends AppCompatActivity {

    private MediaPlayer mMediaPlayer;
    private MediaPlayer.OnCompletionListener mCompletionListener = new MediaPlayer.OnCompletionListener() {
        public void onCompletion(MediaPlayer mp) {
            releaseMediaPlayer();
        }
    };
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.word_list);

        final ArrayList<Word> words = new ArrayList<Word>();

        words.add(new Word("black", "kappu", R.drawable.color_black, R.raw.black));
        words.add(new Word("white", "beeLi", R.drawable.color_white, R.raw.white));
        words.add(new Word("red", "kempu", R.drawable.color_red, R.raw.red));
        words.add(new Word("gray", "buudu", R.drawable.color_gray, R.raw.gray));
        words.add(new Word("green", "hasiru", R.drawable.color_green, R.raw.green));
        words.add(new Word("yellow", "haLadi", R.drawable.color_mustard_yellow, R.raw.yellow));
        words.add(new Word("brown", "kandu", R.drawable.color_brown, R.raw.brown));

        WordAdaptor itemsAdapter = new WordAdaptor(this, words, R.color.category_colors);

        ListView listView = (ListView) findViewById(R.id.list);

        listView.setAdapter(itemsAdapter);

        listView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
                Word word = words.get(position);

                releaseMediaPlayer();
                mMediaPlayer = MediaPlayer.create(ColorsActivity.this, word.getmAudioId());
                mMediaPlayer.start();
                mMediaPlayer.setOnCompletionListener(mCompletionListener);
            }
        });
    }

    protected void onStop(){

        super.onStop();
        releaseMediaPlayer();
    }

    private void releaseMediaPlayer()
    {
        if(mMediaPlayer != null){
            mMediaPlayer.release();
            mMediaPlayer = null;

        }
    }

}

File: app/src/main/java/com/rnadig/learnkannada/ColorsActivity.kt
package com.rnadig.learnkannada

import android.media.MediaPlayer
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.widget.AdapterView
import kotlinx.android.synthetic.main.word_list.*
import java.util.*

/**
 * Created by nsethi on 03-Jul-16.
 */
class ColorsActivity : AppCompatActivity() {

    private var mediaPlayer: MediaPlayer? = null
    private val completionListener = MediaPlayer.OnCompletionListener { releaseMediaPlayer() }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.word_list)

        val words = ArrayList<Word>()
        words.add(Word("black", "kappu", R.drawable.color_black, R.raw.black))
        words.add(Word("white", "beeLi", R.drawable.color_white, R.raw.white))
        words.add(Word("red", "kempu", R.drawable.color_red, R.raw.red))
        words.add(Word("gray", "buudu", R.drawable.color_gray, R.raw.gray))
        words.add(Word("green", "hasiru", R.drawable.color_green, R.raw.green))
        words.add(Word("yellow", "haLadi", R.drawable.color_mustard_yellow, R.raw.yellow))
        words.add(Word("brown", "kandu", R.drawable.color_brown, R.raw.brown))

        val itemsAdapter = WordAdaptor(this, words, R.color.category_colors)
        list.adapter = itemsAdapter
        list.onItemClickListener = AdapterView.OnItemClickListener { parent, view, position, id ->
            val (_, _, _, audioId) = words[position]
            releaseMediaPlayer()
            mediaPlayer = MediaPlayer.create(this@ColorsActivity, audioId)
            mediaPlayer!!.start()
            mediaPlayer!!.setOnCompletionListener(completionListener)
        }
    }

    override fun onStop() {
        super.onStop()
        releaseMediaPlayer()
    }

    private fun releaseMediaPlayer() {
        if (mediaPlayer != null) {
            mediaPlayer!!.release()
            mediaPlayer = null
        }
    }
}